src/main/java/com/conveyal/taui/controllers/BundleController.java

```diff
@@ -204,11 +219,59 @@ public static Bundle update (Request req, Response res) throws IOException {
         return Persistence.bundles.updateFromJSONRequest(req);
     }
 
-    public static Object getBundle (Request req, Response res) {
-        return Persistence.bundles.findByIdFromRequestIfPermitted(req);
+    public static Bundle getBundle (Request req, Response res) {
+        Bundle bundle = Persistence.bundles.findByIdFromRequestIfPermitted(req);
+
+        // Old bundles did not have these set.
+        try {
+            setBundleServiceDates(bundle);
+        } catch (Exception e) {
+            throw AnalysisServerException.unknown(e);
+        }
+
+        return bundle;
+    }
+
+    public static Collection<Bundle> getBundles (Request req, Response res) {
+        return Persistence.bundles.findPermittedForQuery(req);
+    }
+
+    /**
+     * Old bundles were created without computing the service start and end dates. This finds them and stores them.
+     * @param bundle
+     */
+    public static Bundle setBundleServiceDates (Bundle bundle) throws Exception {
+        if (bundle.status != Bundle.Status.DONE || (bundle.serviceStart != null && bundle.serviceEnd != null)) return bundle;
+
+        // Old bundles were created with computing the service start and end dates
+        bundle.serviceStart = LocalDate.MAX;
+        bundle.serviceEnd = LocalDate.MIN;
+
+        for (Bundle.FeedSummary summary : bundle.feeds) {
+            // Compute the feed start and end dates
+            if (summary.serviceStart == null || summary.serviceEnd == null) {
+                FeedSource fs = ApiMain.getFeedSource(Bundle.bundleScopeFeedId(summary.feedId, bundle._id));
+                // Set service start and end from the dates of service
+                List<LocalDate> datesOfService = fs.feed.getDatesOfService();
+                datesOfService.sort(Comparator.naturalOrder());
+                summary.serviceStart = datesOfService.get(0);
+                summary.serviceEnd = datesOfService.get(datesOfService.size() - 1);
+            }
+
+            if (summary.serviceStart.isBefore(bundle.serviceStart)) {
+                bundle.serviceStart = summary.serviceStart;
```

Comment:
This appears to define the service date range as the dates when *any* feed in the bundle has service. Arguably the bundle should only be used where all the feeds overlap, i.e. where *all* feeds in the bundle define service. This would have saved some confusion on various projects where one feed had service defined on only one day, but the other had it defined all year.

Perhaps we should store both date ranges - the range where any feed is active and the range where all feeds are active.

Whatever the definition of this date range is, it should be clearly stated in a comment as a "contract".

Reply:
Updated the comments to reflect the direct usage of these fields.